src/main/java/core/basesyntax/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.operations.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationOperationHandlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationOperationHandlerMap) {
+        this.operationOperationHandlerMap = operationOperationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        return operationOperationHandlerMap.get(operation);
+    }
+}
```

Comment:
It looks like your `OperationStrategyImpl` class is correctly implementing the Strategy pattern by holding a map of operation handlers. However, ensure that you have an appropriate mechanism in place to handle cases where an operation is not supported or is null. It's important to return a meaningful error or throw a specific exception if the `operation` argument passed to the `get` method is not found in the `operationOperationHandlerMap`.